crates/extensions/az_mapper_ext/src/converter.rs

```diff
@@ -48,11 +50,26 @@ impl AzureConverter {
     }
 
     fn try_convert(&mut self, input: &MqttMessage) -> Result<Vec<MqttMessage>, ConversionError> {
+        let default_timestamp = self.add_timestamp.then(|| self.clock.now());
         if is_bridge_health(&input.topic.name) {
             Ok(vec![])
+        } else if input.topic.name.starts_with("tedge/health") {
+            let mut payload_json: Map<String, Value> =
+                serde_json::from_slice(input.payload.as_bytes())?;
+            if let Some(timestamp) = default_timestamp {
+                let timestamp = timestamp
+                    .format(&time::format_description::well_known::Rfc3339)?
+                    .as_str()
+                    .into();
+                payload_json.entry("time").or_insert(timestamp);
+            }
+            let payload = serde_json::to_string(&payload_json)?;
+            Ok(vec![
+                (MqttMessage::new(&self.mapper_config.out_topic, payload)),
+            ])
```

Comment:
Why not simply publish the health status message unchanged?

This piece of code will only add a timestamp to `down` status (the only ones without a `time` field).
And this done using a different time format as for the `up` status (as demonstrated by the `converting_service_health_status` test).

An alternative, is to *rewrite* the `time` field if not using the RFC 3339 format.

Reply:
Yes, we can push the whole `time stamp` without doing any sort of translation.

Reply:
> Why not simply publish the health status message unchanged?
> 
> This piece of code will only add a timestamp to `down` status (the only ones without a `time` field). And this done using a different time format as for the `up` status (as demonstrated by the `converting_service_health_status` test).
> 
> An alternative, is to _rewrite_ the `time` field if not using the RFC 3339 format.

@reubenmiller what's your thought on this? Shall we translate the time to RFC 3339 format or send out the message as it comes in?

Reply:
We need to be consistent, otherwise we'll have the same problem as we have in https://github.com/thin-edge/thin-edge.io/issues/2237.

I would align to what the majority is or if you can find best practises in azure about timestamps. RFC3339 is more human readable but less machine readable.

Reply:
@didier-wenzek The root cause is the API `health_status_up_message(daemon_name: &str) ` where the timestamp is added as a `Unix` timestamp. So, shall we add in `RFC 3339` format while adding a timestamp to the status message? then it will resolve all the issues and no need to re-write while parsing.

Reply:
>  So, shall we add in RFC 3339 format while adding a timestamp to the status message?

Indeed an idea to consider.